app/main.py

```diff
@@ -1,2 +1,48 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(
+            self,
+            capacity: int = 8,
+            load_factor: float = 2 / 3
+    ) -> None:
+        self.capacity = capacity
+        self.load_factor = load_factor
+        self.size = 0
+        self.hash_table = [None] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self._get_index(key)
+        if self.hash_table[index] is None:
+            self.size += 1
+            self.hash_table[index] = (key, hash(key), value)
+        elif self.hash_table[index] is not None:
+            self.hash_table[index] = (key, hash(key), value)
+        if self.size > self.load_factor * self.capacity:
```

Comment:
you should perform this check at the beginning of the method

you ensure that the hash table is always within its load factor limits before inserting a new item